source/mir/math/stat.d

```diff
@@ -180,21 +180,40 @@ unittest
     static assert(is(meanType!(Bar[]) == cfloat));
 }
 
+///
+enum MeanAlgo {
+    /++
+    Computes the sum and the count. The mean is calculated by dividing the sum
+    by the count.
+    +/
+    precise,
+    /++
+    Updates the mean without computing the sum. This is numerically unstable.
+    +/
+    fast
```

Comment:
Instead of a seprate mean accumulator and separate mean kind it is better to have `fastMean` method that uses reciprocal plus multiplication instead of division. New enum and a new struct is too complex and is another thing that using reciprocal. I mean the following:
```
/// mean()
return cast(F) summator.sum / count;

/// fastMean()
return cast(F) summator.sum * (F(1) / count);
```

Just a new small method.

Reply:
> Also, it may make sense to add `twoPass` as a separate algorithm choice for `VarianceAlgo`. In this [paper](https://arxiv.org/pdf/1509.04349.pdf), they find that it can be faster than the `online` algorithm I have added when the data fits in memory.

Looks good